runtime/src/accounts_index.rs

```diff
@@ -23,6 +30,26 @@ pub type RefCount = u64;
 pub type AccountMap<K, V> = BTreeMap<K, V>;
 
 type AccountMapEntry<T> = Arc<AccountMapEntryInner<T>>;
+type SecondaryIndexEntry = DashMap<Pubkey, RwLock<HashSet<Slot>>>;
+
+enum ScanTypes<R: RangeBounds<Pubkey>> {
+    Standard(Option<R>),
+    Indexed(IndexKey),
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum IndexKey {
+    ProgramId(Pubkey),
+    TokenOwner(Pubkey),
+    Mint(Pubkey),
```

Comment:
heh, what a coincidence where all of our current secondary indexes are all indexed by `Pubkey`s. :)